### ISSUE TITLE
The `pkg_resources` package is deprecated

### ISSUE THREAD

**Issue by volks73** [ultralytics/yolov5#13642]
### Search before asking

- [x] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### YOLOv5 Component

Other

### Bug

I recently received the following warning at the terminal:

```
import pkg_resources as pkg
/app/torch/hub/ultralytics_yolov5/utils/general.py:32: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.
```

I checked `utils/general.py` on `main`, and the `pkg_resources` package appears to only be used for the version check function.

Either the `Setuptools` package should be pinned to <81 to the `importlib.resources` and`importlib.metadata` packages used as recommended by the `Setuptools` package's documentation (https://setuptools.pypa.io/en/latest/pkg_resources.html).

### Environment

- YOLOv5 v7.0
- Ubuntu 22.04
- Python v3.11

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @volks73, thank you for your detailed report and interest in YOLOv5 🚀!

This is an automated response to help you get started—an Ultralytics engineer will also assist you soon.

If this is a 🐛 Bug Report, please provide a **minimum reproducible example** (MRE) if possible. This helps us diagnose and resolve the issue more efficiently.

For guidance on setup or best practices, please check our ⭐️ [Tutorials](https://docs.ultralytics.com/yolov5/), including [Custom Data Training](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data/) and [Hyperparameter Evolution](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution/).

## Requirements
[**Python>=3.8.0**](https://www.python.org/) with all [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) installed, including [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/). To get started:
```bash
git clone https://github.com/ultralytics/yolov5  # clone
cd yolov5
pip install -r requirements.txt  # install
```

## Environments
YOLOv5 can be run in any of these up-to-date verified environments (with required dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://bit.ly/yolov5-paperspace-notebook"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

## Status
<a href="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml"><img src="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml/badge.svg" alt="YOLOv5 CI"></a>
If this badge is green, all [YOLOv5 GitHub Actions](https://github.com/ultralytics/yolov5/actions) Continuous Integration (CI) tests are currently passing. CI tests verify correct operation of YOLOv5 [training](https://github.com/ultralytics/yolov5/blob/master/train.py), [validation](https://github.com/ultralytics/yolov5/blob/master/val.py), [inference](https://github.com/ultralytics/yolov5/blob/master/detect.py), [export](https://github.com/ultralytics/yolov5/blob/master/export.py) and [benchmarks](https://github.com/ultralytics/yolov5/blob/master/benchmarks.py) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you for bringing this to our attention! If you’d like to submit a PR, please follow our [contribution guidelines](https://github.com/ultralytics/yolov5/blob/master/CONTRIBUTING.md). We appreciate your help! 🎉

**Comment by volks73** [comment]
Looks like the equivalent of the `pkg_resources.parse_version` is `packaging.version.parse`, which would require adding the `packaging` dependency.

**Comment by pderrenger** [comment]
Thanks for reporting this deprecation warning and offering to help with a PR! You're absolutely right that `pkg_resources` is deprecated and should be replaced. Your suggested approach using `packaging.version.parse` is the correct path forward, and adding the `packaging` dependency is reasonable since it's widely used and lightweight.

While YOLOv5 is now in maintenance mode with active development focused on YOLO11, we definitely welcome important fixes like this one. Please feel free to submit a PR with the `pkg_resources` to `packaging.version.parse` migration - just make sure to update any requirements files accordingly and test that version checking still works as expected.

If you're interested in contributing further, you might also want to check if the same issue exists in the current [YOLO11 codebase](https://github.com/ultralytics/ultralytics) since that's where most new development happens.

**Comment by volks73** [comment]
@pderrenger I have created a PR #13643. I have signed the CLA but the check says it has failed. Maybe someone can re-run the action?

**Comment by pderrenger** [comment]
Thanks for submitting PR #13643! The CLA check sometimes takes a moment to update after signing - I'll trigger a re-run of the checks to see if that resolves it, or if not our team can look into any issues with the CLA verification system.

**Comment by UltralyticsAssistant** [comment]
Thanks for the report 🙏 A potential fix has been merged in the linked PR: Fix deprecation warning for pkg_resources dependency by changing to packaging.

What changed and why this may resolve the issue:
- Replaced deprecated pkg_resources usage with the modern packaging.version API to eliminate deprecation warnings and improve compatibility with newer Python environments.
- Added packaging to pyproject.toml and requirements.txt to ensure the dependency is available.
- Retained setuptools>=70.0.0 for improved security and clarity.
- Cleaned up imports by removing pkg_resources.

Credit to the author and contributors:
- Implemented by volks73 with contributions from glenn-jocher.

How to test the fix:
- Install the latest code from main:
  pip install -U git+https://github.com/ultralytics/yolov5.git@main
- Alternatively, wait for the next official PyPI release and upgrade when available:
  pip install -U ultralytics
- If a specific version is later referenced in this thread or release notes, upgrade to that or newer:
  pip install -U ultralytics>=VERSION

Please let us know if the merged changes resolve the issue on your setup. If the warning persists or you encounter anything else, additional details and logs are welcome.
